### PR TITLE
DO NOT MERGE - SignIn Test

### PR DESCRIPTION
**Pull request by aweng98**
- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by aweng98** [comment]
A experiment with a signin retry.  If any test has retried signin, test log should show whether it worked.

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/fixture/UserFixtures.scala [hunk 1]
```diff
@@ -41,8 +44,18 @@ trait UserFixtures extends CleanUp { self: WebBrowserSpec with TestSuite =>
                                                 (testCode: (T) => Any)
                                                 (implicit webDriver: WebDriver): Unit = {
     withSignIn(user, {
-      new SignInPage(Config.FireCloud.baseUrl).open
-      executeScript(s"window.forceSignedIn('${user.makeAuthToken().value}')")
+      val openedPage: SignInPage = new SignInPage(Config.FireCloud.baseUrl).open
+      retry[T](1.seconds, 60.seconds) ({
+          openedPage.signInButton.awaitVisible()
```
Comment (breilly2) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/fixture/UserFixtures.scala:49]: `SignInPage.open` handles this initially. On retry, sign-in didn't happen so we should be able to assume this still holds. At the very least, we should use `openedPage.awaitReady` instead of widening the visibility of `signInButton`... but I don't think we even need that.

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/PageUtil.scala [hunk 2]
```diff
@@ -17,7 +17,8 @@ trait PageUtil[P <: Page] extends FireCloudView { self: P =>
     * @return the Page object
     */
   def open(implicit webDriver: WebDriver): P = {
-    go to this
+    webDriver.navigate().to(this.url)
```
Comment (breilly2) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/PageUtil.scala:20]: Why this change?

**Comment by aweng98** [comment]
jenkins retest

**Comment by aweng98** [comment]
jenkins retest

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/fixture/UserFixtures.scala [hunk 3]
```diff
@@ -41,8 +44,18 @@ trait UserFixtures extends CleanUp { self: WebBrowserSpec with TestSuite =>
...
+          executeScript(s"window.forceSignedIn('${user.makeAuthToken().value}')")
+          logger.info("Executed JS forceSignedIn")
+          try {
+            await.ready(page, 30)
```
Comment (breilly2) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/fixture/UserFixtures.scala:53]: I'd rather not have to wait 30 seconds if nothing is happening. However, we don't want to cut the wait too short because the workspace list page _could_ be loading.

Maybe instead we can check for the absence of the sign-in button. Something like:

```
await condition (!openedPage.isActive, 5)
```

with this in `SignInPage`:

```
  def isActive: Boolean = signInButton.isVisible
```

That way we retry if the sign-in button doesn't go away within 5 seconds. I haven't tested this so I'm not 100% sure it works.

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/fixture/UserFixtures.scala [hunk 4]
```diff
@@ -41,8 +44,18 @@ trait UserFixtures extends CleanUp { self: WebBrowserSpec with TestSuite =>
                                                 (testCode: (T) => Any)
                                                 (implicit webDriver: WebDriver): Unit = {
     withSignIn(user, {
-      new SignInPage(Config.FireCloud.baseUrl).open
-      executeScript(s"window.forceSignedIn('${user.makeAuthToken().value}')")
+      val openedPage: SignInPage = new SignInPage(Config.FireCloud.baseUrl).open
+      retry[T](1.seconds, 60.seconds) ({
```
Comment (breilly2) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/fixture/UserFixtures.scala:48]: Because this times out after 60 seconds and inside we wait 30 seconds for sign-in to succeed, this will retry approximately 1 time. That's hard logic to follow, somewhat unstable depending on timing, and hard to reason about.

I'd prefer using the other form of `retry`:

```
retry(Seq.fill(2)(1.second)) ({
    ...
```

which inserts two 1-second intervals between tries; this will always try up to 3 times, regardless of how long each try takes.

**Review by breilly2 — CHANGES_REQUESTED** [review]
Let me know if you have questions. Thanks.
